src/frontend/src/components/LinksEditView.tsx

```diff
@@ -0,0 +1,81 @@
+import { useAllLinkTypes } from '../hooks/projects.hooks';
+import LoadingIndicator from './LoadingIndicator';
+import ErrorPage from '../pages/ErrorPage';
+import { Button, Grid, IconButton, MenuItem, Select, TextField } from '@mui/material';
+import { FieldArrayWithId, UseFieldArrayAppend, UseFieldArrayRemove, UseFormRegister, UseFormWatch } from 'react-hook-form';
+import DeleteIcon from '@mui/icons-material/Delete';
+import { getRequiredLinkTypeNames } from '../utils/project.utils';
+import { ProjectEditFormInput } from '../pages/ProjectDetailPage/ProjectEdit/ProjectEditContainer';
+
+const LinksEditView: React.FC<{
+  ls: FieldArrayWithId[];
+  register: UseFormRegister<ProjectEditFormInput>;
+  watch: UseFormWatch<ProjectEditFormInput>;
+  append: UseFieldArrayAppend<any, any>;
+  remove: UseFieldArrayRemove;
+}> = ({ ls, register, append, remove, watch }) => {
+  const { isLoading, isError, error, data: linkTypes } = useAllLinkTypes();
+  if (isLoading || !linkTypes) return <LoadingIndicator />;
+  if (isError) return <ErrorPage message={error.message} />;
+
+  const requiredLinkTypeNames = getRequiredLinkTypeNames(linkTypes);
+
+  const links = watch('links');
+
+  const currentLinkTypeNames = links.map((link) => link.linkTypeName);
+
+  const isRequired = (index: number) => {
```

Comment:
add a comment specifying that this checks whether a link has a required type and has not already appeared in the list of links once